### PR TITLE
Title & Update Title Docs

### PR DESCRIPTION
**Pull request by chriddyp**
These are docs for the new `title=` and `update_title=` features coming in `1.14.0`.

This documentation covers these issues & pull requests:
- https://github.com/plotly/dash/pull/1343
- https://github.com/plotly/dash/pull/1315
- https://github.com/plotly/dash/issues/856
- https://github.com/plotly/dash/issues/732
- https://github.com/plotly/dash-docs/issues/907
- https://github.com/plotly/dash-core-components/issues/833

Post-merge checklist:

The master branch is auto-deployed to `dash.plotly.com`.
Once you have merged your PR, wait 5-10 minutes and check dash.plotly.com
to verify that your changes have been made.

- [x] I understand

If this PR documents a new feature of Dash:

- [ ] Comment on the original Dash issue with a link to the new docs.
- [ ] Reply to any community thread(s) asking for this feature.

If I introduced a new relative link inside `dcc.Markdown`:
- [x] I considered whether I could replace the `dcc.Markdown` call with `rc.Markdown`, which will replace relative links with `tools.relpath` internally. Otherwise, I used e.g. `<dccLink href=tools.relpath('/layout') children="the first chapter"/>` instead of `[the first chapter](/layout)` (importing `tools` from `dash_docs`), and set `dangerously_allow_html=true` in the `dcc.Markdown` call.

### CODE REVIEW

**Comment by chriddyp** [comment]
@jdamiba - Could you review please? Thanks

File: dash_docs/chapters/external_resources/index.py [hunk 1]
```diff
@@ -330,21 +354,96 @@
     ),
 
     rc.Markdown('''
+    ## Customizing Dash's Document or Browser Tab Title
+    The Document Title is the name of the web page that appears in your
+    web browser's tab.
+
+    By default, it is Dash.
+
+    In Dash 1.13.5, you can customize this title with the `title=` keyword:
```
Comment (jdamiba) [inline dash_docs/chapters/external_resources/index.py:363]:
```suggestion
    As of Dash 1.13.5, you can customize this title with the `title=` keyword:
```

Comment (chriddyp) [reply dash_docs/chapters/external_resources/index.py:363]: 🦅 👀  Much better! Thanks @jdamiba

File: dash_docs/chapters/external_resources/index.py [hunk 2]
```diff
@@ -290,6 +291,29 @@
         style=styles.code_container
     ),
 
+    rc.Markdown(
+    '''
+    > If placing images inside the `assets` folder isn't an option, then you can
+    > you can also embed images "inline" with base64 encoding:
```
Comment (Marc-Andre-Rivet) [inline dash_docs/chapters/external_resources/index.py:297]:
```suggestion
    > also embed images "inline" with base64 encoding:
```

File: dash_docs/chapters/external_resources/index.py [hunk 3]
```diff
@@ -330,21 +354,96 @@
...
+    ```
+
+    > Note - Prior to 1.14.0, setting `app.title` was an unnofficial way to
+    > set this title. This is still possible but may be removed in the future.
+    > We recommend using `title=` instead.
+
+    ## Update the Document Title Dynamically based off of the URL or Tab
+
+    To set the document title dynamically, you can use a clientside callback
+    that updates the `document.title` as a side affect. The example below
```
Comment (Marc-Andre-Rivet) [inline dash_docs/chapters/external_resources/index.py:375]:
```suggestion
    that updates the `document.title` as a side effect. The example below
```

Comment (chriddyp) [reply dash_docs/chapters/external_resources/index.py:375]: good catch!

File: dash_docs/chapters/external_resources/index.py (hunk 2 above)
Comment (chriddyp) [reply dash_docs/chapters/external_resources/index.py:297]: Yes We Can!

**Pull request merged**